### PR TITLE
HPCC-17478 Number of partitions should be configurable in Cassandra WU store

### PR DESCRIPTION
**Pull request by richardkchapman**
Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

## Testing:
Ran workunit seftests in wutools. Found and fixed some missing overrides. Checked version table looks as it should.

### CODE REVIEW

**Review by mckellyln — APPROVED** [review]
@richardkchapman changes look good,

**Comment by richardkchapman** [comment]
@ghalliday Please review/merge

File: plugins/cassandra/cassandrawu.cpp [hunk 1]
```diff
@@ -3126,6 +3153,8 @@ class CCasssandraWorkUnitFactory : public CWorkUnitFactory, implements ICassandr
             {
                 int major = versionInfo->getPropInt("@major", 0);
                 int minor = versionInfo->getPropInt("@minor", 0);
+                partitions = versionInfo->getPropInt("@numPartitions", DEFAULT_PARTITIONS);
```
Comment (ghalliday) [inline plugins/cassandra/cassandrawu.cpp]: @richardkchapman are these values saved anywhere?  I don't think they're in the version mappings.

Comment (richardkchapman) [inline plugins/cassandra/cassandrawu.cpp]: They are stored in attributes column:

    {"attributes", "map<text, text>", "@major@minor@partition@", attributeMapColumnMapper},  // name is the suppression list, note trailing @

**Comment by richardkchapman** [comment]
@ghalliday Back to you

File: plugins/cassandra/cassandrawu.cpp (hunk 1 above)
Comment (ghalliday) [inline plugins/cassandra/cassandrawu.cpp]: I found it - what I was missing was the line:

VStringBuffer versionInfo....

**Pull request merged**